### PR TITLE
Polish: storage docs

### PR DESCRIPTION
**Pull request by EndymionJkb**
# Description

Guess I missed this during the Vault polish. Some comments were out-of-date, and things were fairly jumbled around. This separates things and adds block comments to make everything more readable.

The diff isn't very legible. Best to just view the current file; there are no code changes.

Interestingly, just moving things around seemed to increase bytecode and decrease gas. This makes some sense, because it does lay the storage out in order. Maybe there is an optimal order for variables to maximize gas, even though it would be a small effect.

If it's large enough to worry about, we could make this the VaultStorageDoc interface for readability, but not use it. It would have a comment pointing to the optimized VaultStorage, which would just declare everything in optimal order with no comments.

## Type of change

- [ ] Bug fix 
- [ ] New feature 
- [ ] Breaking change 
- [ ] Dependency changes
- [ ] Code refactor / cleanup
- [ ] Optimization: [ ] gas / [ ] bytecode
- [ ] Documentation or wording changes
- [ ] Other

## Checklist:

- [ ] The diff is legible and has no extraneous changes
- [ ] Complex code has been commented, including external interfaces
- [ ] Tests have 100% code coverage
- [ ] The base branch is either `main`, or there's a description of how to merge

### CODE REVIEW

File: pkg/vault/test/.contract-sizes/Vault [hunk 1]
```diff
@@ -1,2 +1,2 @@
-Bytecode	23.816
-InitCode	25.092
\ No newline at end of file
+Bytecode	23.943
```
Comment (jubeira) [inline pkg/vault/test/.contract-sizes/Vault:1]: ooof ~120 bytes for 100 gas? Are we sure this is from this PR?

I was hoping to get some bytecode buffer to play a bit with the compiler settings; feels like we can get better gas improvements out of that.

We also have some external getters in the extension that would be better off in the main vault for the batch router.

**Review by jubeira — COMMENTED** [review]
LGTM; I'm a bit concerned about the bytecode size increase. Are we sure it's from this PR?

File: pkg/vault/test/.contract-sizes/Vault (hunk 1 above)
Comment (EndymionJkb) [reply pkg/vault/test/.contract-sizes/Vault]: It seems to be from this PR, as I can influence it by changing the order. Experimenting now with shifting things around.

Comment (EndymionJkb) [reply pkg/vault/test/.contract-sizes/Vault]: It really doesn't like moving the contract addresses

**Review by jubeira — APPROVED** [review]
LGTM, thanks for investigating this! :)

**Pull request merged**